### DISCUSSION TITLE
Multiple table columns extracted as one column

### DISCUSSION THREAD

**Discussion by malek-ewa** [jsvine/pdfplumber#859] [Ask for help with specific PDFs]
Hey there,

I am trying to extract data from a table, my code looks like:

```
table = page.extract_table(table_settings={
    "horizontal_strategy": "text",
    "vertical_strategy": "text",
    "keep_blank_chars" : True
})
```

The issue I'm having is that this seems to be extracting some columns as one: my PDF looks like this:

![Image](https://user-images.githubusercontent.com/96867680/231414635-f68bbbfd-cd43-457c-b7a7-9484f8623442.png)

But the data that's extracted has this structure:

![Image](https://user-images.githubusercontent.com/96867680/231417126-5807524e-6cff-4f6f-aab9-5d14c3375f67.png)

So basically, columns 1, 2 and 3 are being extracted as one column, and then the last 2 columns are also being extracted as one column.

So my question is: which parameters / settings would you recomment that I adjust, and to which values shall I adjust them, in order to get those columns to be extracted as individual columns ?

I already tried different values for `join_tolerance`,  `snap_tolerance` and `intersection_tolerance`,  but couldn't get it to work.

Thanks in advance.
Malek

**Comment by jsvine** [comment]
Yes, unfortunately the `text` strategies can be somewhat harder to finesse, especially with "sparse" columns (where relatively few of the cells are filled in) like in the example in your screenshot. In cases like these, you might be have more luck using `"vertical_strategy": "explicit"` and `"explicit_vertical_lines": [...]`, where those explicit lines are either hardcoded by you (if you expect the columns to appear in the same positions on each page), or derived programmatically from other features (text/graphics) on the page.

**Reply by malek-ewa** [reply]
Understood. I'll give that a try. Thanks a lot for your reply @jsvine

**Reply by malek-ewa** [reply]
Hi again @jsvine 

The vertical strategy does not seem to be easy to get to work in my case, because the positions of the columns might vary from a table to another. So I was trying to see if I can get it to work by using extract_text with layout True, and I have a question regarding that.

My file looks like this:

![Image](https://user-images.githubusercontent.com/96867680/231669276-63e9c8d8-ec9b-4733-9f29-dccef3076ff2.png)

`extract_text()` with the default arguments is working fine - my data looks like this in that case:

![Image](https://user-images.githubusercontent.com/96867680/231669570-5a10e824-ff1e-42ed-a462-9789087bdab0.png)

However, when I use `extract_text(layout=True)`, the different lines get weirdly merged for the first column - basically the dates and the timestamps are parsed as one value, for a given line with the line that's just before:

![Image](https://user-images.githubusercontent.com/96867680/231670402-d6e63fd7-f93b-4aea-9728-12b8f247859f.png)

Is there anything I could do to fix that ?

Thanks in advance.

**Reply by jsvine** [reply]
It's a bit hard to know without having access to the PDF itself. But two possibilities:

- Upgrade to the newly-released `v0.9.0` and rerun. I ironed out a few small edge-cases there that might help.
- Try tweaking the `y_tolerance` parameter to `.extract_text(...)`.

If neither work, and you're unable to share the PDF, can you share the results of this?:

```python
im = page.to_image()
im.draw_rects(page.chars)
im.save("test.png")
```
